File: 0311-Sparse Matrix Multiplication/0311-Sparse Matrix Multiplication.cpp
class Solution {
public:
    vector<vector<int>> multiply(vector<vector<int>>& A, vector<vector<int>>& B) {
        int m = A.size(), n = A[0].size(), p = B[0].size();
        vector<vector<int>> result(m, vector<int>(p));
        for (int i = 0; i < m; ++i)
        {
            for (int k = 0; k < n; ++k)
            {
                if (A[i][k])
                {
                    for (int j = 0; j < p; ++j)
                    {
                        result[i][j] += A[i][k] * B[k][j];
                    }
                }
            }
        }
        
        return result;
    }
};

File: 0301-0400/0311-Sparse Matrix Multiplication/0311-Sparse Matrix Multiplication.py
class Solution:
    def multiply(self, A: List[List[int]], B: List[List[int]]) -> List[List[int]]:
        m, n, p = len(A), len(A[0]), len(B[0])
        result = [[0] * p for _ in range(m)]
        for i in range(m):
            for j in range(n):
                if A[i][j]:
                    for k in range(p):
                        result[i][k] += A[i][j] * B[j][k]
        return result
